src/runtimes/node/detect_es_module.js

```diff
@@ -0,0 +1,25 @@
+const { readFile } = require('fs')
+const { promisify } = require('util')
+
+const pReadFile = promisify(readFile)
+
+const { init, parse } = require('es-module-lexer')
+
+const detectEsModule = async ({ mainFile }) => {
+  if (!mainFile) {
+    return false
+  }
+
+  try {
+    const [mainFileContents] = await Promise.all([pReadFile(mainFile, { encoding: 'utf8' }), init])
```

Comment:
Tiny detail, just for consistency with the rest of the codebase:

```suggestion
    const [mainFileContents] = await Promise.all([pReadFile(mainFile, 'utf8'), init])
```